service/resource/project/resource.go

```diff
@@ -0,0 +1,191 @@
+package project
+
+import (
+	"fmt"
+
+	"github.com/giantswarm/draughtsmantpr"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/framework"
+)
+
+const (
+	// Name is the identifier of the resource.
+	Name = "project"
+)
+
+// Config represents the configuration used to create a new project resource.
+type Config struct {
+	// Dependencies.
+	Logger micrologger.Logger
+}
+
+// DefaultConfig provides a default configuration to create a new project
+// resource by best effort.
+func DefaultConfig() Config {
+	return Config{
+		// Dependencies.
+		Logger: nil,
+	}
+}
+
+// Resource implements the project resource.
+type Resource struct {
+	// Dependencies.
+	logger micrologger.Logger
+}
+
+// New creates a new configured project resource.
+func New(config Config) (*Resource, error) {
+	// Dependencies.
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+
+	newService := &Resource{
+		// Dependencies.
+		logger: config.Logger.With(
+			"resource", Name,
+		),
+	}
+
+	return newService, nil
+}
+
+func (r *Resource) GetCurrentState(obj interface{}) (interface{}, error) {
+	customObject, err := toCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.Log("debug", "get current state")
+
+	r.logger.Log("TODO", fmt.Sprintf("implement logic based on received custom object: %#v", customObject))
```

Comment:
This is the new project resource we still need to implement. I want to wire the eventer and the operator first and then proceed with the business logic implementation in the next step.